ctapipe/calib/camera/charge_extraction.py

```diff
@@ -0,0 +1,281 @@
+from abc import abstractmethod
+import numpy as np
+from traitlets import Int, Unicode
+
+from ctapipe.core import Component
+
+
+def all_subclasses(cls):
+    return cls.__subclasses__() + [g for s in cls.__subclasses__()
+                                   for g in all_subclasses(s)]
+
+
+class ChargeExtractor(Component):
```

Comment:
You need to pass `metaclass=ABCMeta` to the class definition to be able to use @abstractmethod.